### PR TITLE
fix: Ensure async component options in form schemas are not applied to destroyed components

### PR DESCRIPTION
**Pull request by Richard-Walton**
Can happen if the form is destroyed before a component's options promise had resolved.   Again, this was noticed yesterday when my internet was being very slow.

The diff in question starts here: https://github.com/milojs/milo-ui/compare/fix/async-racecondition?expand=1#diff-8bea722f23530356b725cc0cae78bd0bR122

### CODE REVIEW

File: lib/forms/item_types.js [hunk 1]
```diff
@@ -119,16 +119,21 @@ function processInputSchema(comp, schema) {
 }
 
 function setComponentOptions(comp, options, setModelFunc) {
+    function trySetModelFunc(comp, data) {
+        if (! comp.isDestroyed()) setModelFunc(comp, data);
+    };
```
Comment (maael) [inline lib/forms/item_types.js]: Don't need the semi colon on the end of a function definition.

Comment (maael) [reply lib/forms/item_types.js]: Also is it a convention that you follow where you put a space between `!` and the variable? I've never seen it before outside of a few places in CC. What's the reasoning for it, or is it just a preference?

File: lib/forms/item_types.js [hunk 2]
```diff
@@ -119,16 +119,21 @@ function processInputSchema(comp, schema) {
...
+
     if (options) {
         if (typeof options.then == 'function') {
-            setModelFunc(comp, [{ value: 0, label: 'loading...' }]);
+            trySetModelFunc(comp, [{ value: 0, label: 'loading...' }]);
             options
                 .then(
-                    function(data) { setModelFunc(comp, data); },
-                    function() { setModelFunc(comp, [{ value: 0, label: 'loading error' }]); }
+                    function(data) { trySetModelFunc(comp, data); },
```
Comment (maael) [inline lib/forms/item_types.js:131]: Any reason you're not going for arrow functions here, as it's a nice place for the shorter syntax and you're not working with the context at all.

**Review by maael — DISMISSED** [review]
Added a comment about a semi colon, and some questions on code styling. Nothing blocking it from being merged however.

File: lib/forms/item_types.js (hunk 1 above)
Comment (Richard-Walton) [reply lib/forms/item_types.js]: 1) - Oops, no idea how he snuck in there.  I removed
2) Can't say I've ever given it much thought.  :)

File: lib/forms/item_types.js (hunk 2 above)
Comment (Richard-Walton) [reply lib/forms/item_types.js:131]: The original implementation used `function` syntax.  I didn't want to add 'noise' to this PR.

**Review by maael — APPROVED** [review]
👍 My curiosity has been satisfied, you're good to go.

**Review by jasoniangreen — APPROVED** [review]

**Pull request merged**